File: app/OnyxPenDemo/src/main/java/com/onyx/android/eink/pen/demo/event/PenEvent.java
package com.onyx.android.eink.pen.demo.event;

import static com.onyx.android.sdk.data.note.NoteConstant.COLOR_DEVICE_PEN_RESUME_DELAY_TIME_MS;
import static com.onyx.android.sdk.data.note.NoteConstant.COMMON_PEN_RESUME_DELAY_TIME_MS;

import com.onyx.android.sdk.utils.DeviceInfoUtil;

public class PenEvent {
    public static final int DELAY_ENABLE_RAW_DRAWING_MILLS = DeviceInfoUtil.isColorDevice() ?
            COLOR_DEVICE_PEN_RESUME_DELAY_TIME_MS : COMMON_PEN_RESUME_DELAY_TIME_MS;
    public static final int POPUP_RESUME_PEN_TIME_MS = DeviceInfoUtil.isColorDevice() ? 500 : 300;
    private boolean resumeDrawingRender;
    private boolean resumeRawInputReader;
    private int delayResumePenTimeMs = DELAY_ENABLE_RAW_DRAWING_MILLS;

    public PenEvent(boolean resumeDrawingRender, boolean resumeRawInputReader, int delayResumePenTimeMs) {
        this.resumeDrawingRender = resumeDrawingRender;
        this.resumeRawInputReader = resumeRawInputReader;
        this.delayResumePenTimeMs = delayResumePenTimeMs;
    }

    public static PenEvent resumeRawDrawing(int delayResumePenTimeMs) {
        return new PenEvent(true, true, delayResumePenTimeMs);
    }

    public boolean isResumeDrawingRender() {
        return resumeDrawingRender;
    }

    public boolean isResumeRawInputReader() {
        return resumeRawInputReader;
    }

    public int getDelayResumePenTimeMs() {
        return delayResumePenTimeMs;
    }
}

File: app/OnyxPenDemo/src/main/java/com/onyx/android/eink/pen/demo/event/PenEvent.kt
package com.onyx.android.eink.pen.demo.event

import com.onyx.android.sdk.data.note.NoteConstant
import com.onyx.android.sdk.utils.DeviceInfoUtil

class PenEvent(
    private val resumeDrawingRender: Boolean,
    private val resumeRawInputReader: Boolean,
    delayResumePenTimeMs: Int
) {
    private var delayResumePenTimeMs: Int = DELAY_ENABLE_RAW_DRAWING_MILLS

    init {
        this.delayResumePenTimeMs = delayResumePenTimeMs
    }

    fun isResumeDrawingRender(): Boolean = resumeDrawingRender

    fun isResumeRawInputReader(): Boolean = resumeRawInputReader

    fun getDelayResumePenTimeMs(): Int = delayResumePenTimeMs

    companion object {
        val DELAY_ENABLE_RAW_DRAWING_MILLS: Int =
            if (DeviceInfoUtil.isColorDevice()) {
                NoteConstant.COLOR_DEVICE_PEN_RESUME_DELAY_TIME_MS
            } else {
                NoteConstant.COMMON_PEN_RESUME_DELAY_TIME_MS
            }
        val POPUP_RESUME_PEN_TIME_MS: Int = if (DeviceInfoUtil.isColorDevice()) 500 else 300

        /** Stop Soft Eraser overlay after side-button / cap erase while brush stays selected.  */
        @JvmStatic
        fun pauseDrawingRender(): PenEvent {
            return PenEvent(false, false, 0)
        }

        @JvmStatic
        fun resumeRawDrawing(delayResumePenTimeMs: Int): PenEvent {
            return PenEvent(true, true, delayResumePenTimeMs)
        }

        @JvmStatic
        fun resumeRawDrawingImmediately(): PenEvent {
            return PenEvent(true, true, 0)
        }
    }
}
